apps/full-site-editing/full-site-editing-plugin/wp-template.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+/**
+ * Mostly a copy of how the `wp_block` is registered in core
+ *
+ * @return void
+ */
+function fse_register_wp_template() {
+	register_post_type(
+		'wp_template',
+		array(
+			'labels'                => array(
+				'name'                     => _x( 'Templates', 'post type general name' ),
+				'singular_name'            => _x( 'Template', 'post type singular name' ),
+				'menu_name'                => _x( 'Templates', 'admin menu' ),
+				'name_admin_bar'           => _x( 'Template', 'add new on admin bar' ),
+				'add_new'                  => _x( 'Add New', 'Template' ),
+				'add_new_item'             => __( 'Add New Template' ),
+				'new_item'                 => __( 'New Template' ),
+				'edit_item'                => __( 'Edit Template' ),
+				'view_item'                => __( 'View Template' ),
+				'all_items'                => __( 'All Templates' ),
+				'search_items'             => __( 'Search Templates' ),
+				'not_found'                => __( 'No Templates found.' ),
+				'not_found_in_trash'       => __( 'No Templates found in Trash.' ),
+				'filter_items_list'        => __( 'Filter Templates list' ),
+				'items_list_navigation'    => __( 'Templates list navigation' ),
+				'items_list'               => __( 'Templates list' ),
+				'item_published'           => __( 'Template published.' ),
+				'item_published_privately' => __( 'Template published privately.' ),
+				'item_reverted_to_draft'   => __( 'Template reverted to draft.' ),
+				'item_scheduled'           => __( 'Template scheduled.' ),
+				'item_updated'             => __( 'Template updated.' ),
+			),
+			'public'                => true, // temp to support the search endpoint
+			'show_ui'               => true,
+			'show_in_menu'          => true,
+			'rewrite'               => false,
+			'show_in_rest'          => true,
+			'rest_base'             => 'templates',
+			'rest_controller_class' => 'A8C_REST_Templates_Controller',
+			'capability_type'       => 'Template',
+			'capabilities'          => array(
+				// You need to be able to edit posts, in order to read Templates in their raw form.
+				'read'                   => 'edit_posts',
+				// You need to be able to publish posts, in order to create Templates.
+				'create_posts'           => 'customize',
+				'edit_posts'             => 'customize',
+				'edit_published_posts'   => 'customize',
+				'delete_published_posts' => 'customize',
+				'edit_others_posts'      => 'customize',
+				'delete_others_posts'    => 'customize',
+			),
+			'map_meta_cap'          => true,
+			'supports'              => array(
+				'title',
+				'editor',
```

Comment:
What about adding `revisions`, and maybe also `custom-fields` (probably not needed _now_, but I don't see a reason to actually prevent it)?

Reply:
I think let's turn those on when/if needed. These are copied straight from `wp_block` and if they didn't need more there, we don't either until we know we do.